src/components/CsvImport/csv.js

```diff
@@ -42,13 +43,22 @@ export default {
         worker: true,
         comments: false,
         step: undefined,
-        complete: results => {
+        chunk: results => {
+          if (Object.keys(parsedData).length === 0 && parsedData.constructor === Object) {
```

Comment:
> `parsedData.constructor === Object`

This condition looks unnecessary. Does `parsedData` have a chance to change its type?
How about initialization `parsedData` with `null` instead of `{}` so we could just check `parsedData === null` for the first chunk?

Reply:
I was trying to clearly identify it as an object, but certainly this makes it more readable.